File: Free Html To Pdf Converter for ASP.NET MVC/[C#]-Free Html To Pdf Converter for ASP.NET MVC/C#/SelectPdf.Samples/Controllers/HttpPostRequestController.cs
using System;
using System.Web.Mvc;

namespace SelectPdf.Samples.Controllers
{
    public class HttpPostRequestController : Controller
    {
        // GET: HttpPostRequest
        public ActionResult Index()
        {
            string url = System.Web.VirtualPathUtility.ToAbsolute("~/ViewHttpPostData");
            ViewData.Add("ViewTxtUrl", (new Uri(Request.Url, url)).AbsoluteUri);
            return View();
        }

        [HttpPost]
        public ActionResult SubmitAction(FormCollection fields)
        {
            // instantiate a html to pdf converter object
            HtmlToPdf converter = new HtmlToPdf();

            // set the HTTP POST parameters
            converter.Options.HttpPostParameters.
                Add(fields["TxtName1"], fields["TxtValue1"]);
            converter.Options.HttpPostParameters.
                Add(fields["TxtName2"], fields["TxtValue2"]);
            converter.Options.HttpPostParameters.
                Add(fields["TxtName3"], fields["TxtValue3"]);
            converter.Options.HttpPostParameters.
                Add(fields["TxtName4"], fields["TxtValue4"]);

            // create a new pdf document converting an url
            PdfDocument doc = converter.ConvertUrl(fields["TxtUrl"]);

            // save pdf document
            byte[] pdf = doc.Save();

            // close pdf document
            doc.Close();

            // return resulted pdf document
            FileResult fileResult = new FileContentResult(pdf, "application/pdf");
            fileResult.FileDownloadName = "Document.pdf";
            return fileResult;
        }
    }
}

File: Free Html To Pdf Converter for ASP.NET MVC/[VB]-Free Html To Pdf Converter for ASP.NET MVC/VB/SelectPdf.Samples/Controllers/HttpPostRequestController.vb
Imports System.Web
Imports SelectPdf

Namespace Controllers
    Public Class HttpPostRequestController
        Inherits Controller

        ' GET: HttpPostRequest
        Function Index() As ActionResult
            Dim url As String = VirtualPathUtility.ToAbsolute("~/ViewHttpPostData")
            ViewData.Add("ViewTxtUrl", (New Uri(Request.Url, url)).AbsoluteUri)
            Return View()
        End Function

        <HttpPost>
        Public Function SubmitAction(fields As FormCollection) As ActionResult
            ' instantiate a html to pdf converter object
            Dim converter As New HtmlToPdf()

            ' set the HTTP POST parameters
            converter.Options.HttpPostParameters.Add(
                fields("TxtName1"), fields("TxtValue1"))
            converter.Options.HttpPostParameters.Add(
                fields("TxtName2"), fields("TxtValue2"))
            converter.Options.HttpPostParameters.Add(
                fields("TxtName3"), fields("TxtValue3"))
            converter.Options.HttpPostParameters.Add(
                fields("TxtName4"), fields("TxtValue4"))

            ' create a new pdf document converting an url
            Dim doc As PdfDocument = converter.ConvertUrl(fields("TxtUrl"))

            ' save pdf document
            Dim pdf As Byte() = doc.Save()

            ' close pdf document
            doc.Close()

            ' return resulted pdf document
            Dim fileResult As FileResult = New FileContentResult(pdf, "application/pdf")
            fileResult.FileDownloadName = "Document.pdf"
            Return fileResult
        End Function
    End Class
End Namespace